include/zephyr/bluetooth/audio/csis.h

```diff
@@ -93,6 +96,15 @@ struct bt_csis_cb {
 	 * @return A BT_CSIS_READ_SIRK_REQ_RSP_* response code.
 	 */
 	uint8_t (*sirk_read_req)(struct bt_conn *conn, struct bt_csis *csis);
+
+	/**
+	 * @brief Callback whenever the RSI changes.
+	 *
+	 * If this callback is not set, the stack will handle advertising of the RSI.
+	 *
+	 * @param rsi Pointer to the new 6-octet RSI data to be advertised.
+	 */
+	void (*rsi_changed)(const uint8_t rsi[]);
```

Comment:
We might actually be to do this
```suggestion
	void (*rsi_changed)(const uint8_t rsi[BT_CSIS_RSI_SIZE]);
```
To provide information about it being a fixed size array.